docusaurus/docs/dev-docs/plugins/graphql.md

```diff
@@ -903,3 +903,89 @@ Using API tokens in the the GraphQL playground requires adding the authorization
 
 Replace `<TOKEN>` with your API token generated in the Strapi Admin panel.
 :::
+
+## Security
+
+As GraphQL is a query language, it allows users to use a wider panel of inputs than traditional REST APIs.
+
+Due to this feature, GraphQL APIs are inherently prone to various security risks, but they can be reduced by taking appropriate precautions.
+Neglecting them can expose the API to vulnerabilities like credential leakage or denial of service attacks.
+
+### Disable introspection and playground in production
+
+In production, it is recommended to disable the GraphQL Playground and the introspection query.
+If you haven't edited the [configuration file](/dev-docs/configurations/plugins#graphql-configuration), it is already disabled in production by default.
+
+### Limit max depth and complexity
+
+A malicious user could send a query with a very high depth, which could overload your server. Therefore, a depth limit can be set to limit the maximum number of nested fields that can be queried in a single request.
+
+Depending on your use case, such as during testing and development, you can set the `depthLimit` option to a higher value.
+
+Review the [configuration file](/dev-docs/configurations/plugins#graphql-configuration) to set this options.
+
+### Install graphql armor
```

Comment:
:::tip
To increase GraphQL security even further, 3rd-party tools can be used. See the guide about [using GraphQL Armor with Strapi on the forum](https://forum.strapi.io/t/use-graphql-armor-with-strapi/).
:::